modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/descriptor/AbstractArtifactDescriptorFactory.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.descriptor;
+
+import org.mule.runtime.api.deployment.meta.AbstractMuleArtifactModel;
+import org.mule.runtime.api.meta.MuleVersion;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptorValidator;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptorValidatorBuilder;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.ClassLoaderModel;
+
+import java.io.File;
+
+/**
+ * Base class to create artifact descriptors.
+ *
+ * @param <M> type of the artifact model that owns the descriptor.
+ * @param <T> type of descriptor being created.
+ */
+public abstract class AbstractArtifactDescriptorFactory<M extends AbstractMuleArtifactModel, T extends ArtifactDescriptor> {
+
+  protected final File artifactLocation;
+  protected final M artifactModel;
```

Comment:
private

Reply:
Both variables are used in child classes.